tests/test_optimizer.py

```diff
@@ -983,6 +983,97 @@ def test_constrained_AL_scalar():
     np.testing.assert_array_less(-Dwell, ctol)
 
 
+@pytest.mark.slow
+@pytest.mark.unit
+@pytest.mark.optimize
+def test_signed_PlasmaVesselDistance():
+    """Tests that signed distance works with surface optimization."""
+    eq = desc.examples.get("HELIOTRON")
+
+    constraints = (FixParameters(eq),)  # don't want eq to change
+    surf = eq.surface.copy()
+    surf.change_resolution(M=1, N=1)
+
+    grid = LinearGrid(M=eq.M, N=eq.N, NFP=eq.NFP)
+    obj = PlasmaVesselDistance(
+        surface=surf,
+        eq=eq,
+        target=-0.25,
+        surface_grid=grid,
+        plasma_grid=grid,
+        use_signed_distance=True,
+    )
+    objective = ObjectiveFunction((obj,))
+
+    optimizer = Optimizer("lsq-exact")
+    (eq, surf), _ = optimizer.optimize(
+        (eq, surf), objective, constraints, verbose=3, maxiter=60, ftol=1e-8, xtol=1e-9
+    )
+
+    np.testing.assert_allclose(
+        obj.compute(*obj.xs(eq, surf)), -0.25, atol=1e-2, err_msg="Using hardmin"
+    )
+
+    # with softmin
```

Comment:
Do you mind explaining what softmin is? Or is there a resource with a good definition?

Reply:
https://en.wikipedia.org/wiki/Smooth_maximum

https://github.com/PlasmaControl/DESC/blob/8c4051a555365f87c34871cf073a35f2c0d06e93/desc/objectives/utils.py#L215

basically is a way to smoothly approximate the max/min function, where the parameter alpha determines the degree of approximation, for large alpha the functions tend to yield the max value of the input array. This is useful as it allows the gradient of the output of this function to be nonzero for every entry, not just for the entry that is the largest/smallest (which is what the max/min function would do, as the function ONLY depends on whatever the max/min value is and not on the other parts of the array)